src/plugins/vis_builder/public/application/components/data_tab/index.tsx

```diff
@@ -3,19 +3,265 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { DropResult, EuiDragDropContext } from '@elastic/eui';
 import { FieldSelector } from './field_selector';
 
 import './index.scss';
 import { ConfigPanel } from './config_panel';
+import { useAggs, useVisualizationType } from '../../utils/use';
+import { useTypedDispatch, useTypedSelector } from '../../utils/state_management';
+
+import {
+  reorderAgg,
+  updateAggConfigParams,
+} from '../../utils/state_management/visualization_slice';
+import { useOpenSearchDashboards } from '../../../../../opensearch_dashboards_react/public';
+import { VisBuilderServices } from '../../../types';
+import { createNewAggConfig } from './utils/get_valid_aggregations';
+import { DropboxDisplay } from './dropbox';
 
 export const DATA_TAB_ID = 'data_tab';
 
+export interface SchemaDisplayStates {
+  [key: string]: DropboxDisplay[];
+}
+
 export const DataTab = () => {
+  // Field Selector panel States
+  const fieldSelectorGroups = [
+    'preDefinedCountMetric',
+    'categoricalFields',
+    'numericalFields',
+    'metaFields',
+  ];
```

Comment:
You seem to be referencing these string in other places. Can you make them constants and reference them in other places instead of performing string matching?

Reply:
Consolidated these into an enum and using it across files.